infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -263,15 +287,45 @@ public static void main(String[] args) throws Exception {
             throw new IllegalArgumentException("Max number of metadata files to retain must be greater than 0.");
         }
 
-        corfuServerThread = new Thread(() -> startServer(opts, bindToAllInterfaces));
-        corfuServerThread.setName("CorfuServer");
-        corfuServerThread.start();
+        // Register shutdown handler
+        Thread shutdownThread = new Thread(CorfuServer::cleanShutdown);
+        shutdownThread.setName("ShutdownThread");
+        Runtime.getRuntime().addShutdownHook(shutdownThread);
+
+        while (!SHUTDOWN_SERVER) {
+            final ServerContext sc = new ServerContext(opts);
+            try (CorfuServer corfuServer = new CorfuServer(sc)) {
```

Comment:
`ACTIVE_SERVER.start().channel().closeFuture().syncUninterruptibly();`  already implies that `CorfuServer::close()` is called and also `System.exit(EXIT_ERROR_CODE)` will invoke shutdownHook and also call `CorfuServer::close()`. So you don't need to use try with resource, which issues an extra close(), right?